Sources/Maintenance/Tools/Install.php

```diff
@@ -1503,6 +1525,30 @@ private function createImageProxySecret(): string
 		return bin2hex(random_bytes(10));
 	}
 
+	/**
+	 * Get our upgrade data.
+	 */
+	private function getUpgradeData(): void
+	{
+		$defined_vars = Config::getCurrentSettings();
+
+		$data = isset($defined_vars['upgradeData']) ? Utils::jsonDecode($defined_vars['upgradeData'], true) : [];
+
+		$this->time_started = isset($data['started']) ? (int) $data['started'] : time();
+	}
+
+	/**
+	 * Save our data.
+	 *
+	 * @return bool True if we could update our settings file, false otherwise.
+	 */
+	private function saveUpgradeData(): bool
+	{
+		return Config::updateSettingsFile(['upgradeData' => json_encode([
+			'started' => $this->time_started,
+		])]);
+	}
+
```

Comment:
Technically should be called save/get InstallData, but doesn't mater.  Didn't even think of yet that the installer now has a timer since it is getting it from the upgrader.

Reply:
Yeah, I decided to just stick with "upgradeData" in order to avoid having to add something else to check for in SMF\Config.